internal/v2/fakes2av2/fakes2av2_test.go

```diff
@@ -0,0 +1,177 @@
+					&s2av2pb.GetTlsConfigurationReq {
+						ConnectionSide: commonpb.ConnectionSide_CONNECTION_SIDE_CLIENT,
+					},
+				},
+			},
+			expectedResponse: &s2av2pb.SessionResp {
+				Status: &s2av2pb.Status {
+					Code: 0,
+				},
+				RespOneof: &s2av2pb.SessionResp_GetTlsConfigurationResp {
+					GetTlsConfigurationResp: &s2av2pb.GetTlsConfigurationResp {
+						TlsConfiguration: &s2av2pb.GetTlsConfigurationResp_ClientTlsConfiguration_ {
+							&s2av2pb.GetTlsConfigurationResp_ClientTlsConfiguration {
+								CertificateChain: []string{
+									string(clientCert),
+								},
+								MinTlsVersion: commonpb.TLSVersion_TLS_VERSION_1_3,
+								MaxTlsVersion: commonpb.TLSVersion_TLS_VERSION_1_3,
+								HandshakeCiphersuites: []commonpb.HandshakeCiphersuite{},
+								RecordCiphersuites: []commonpb.RecordCiphersuite {
+									commonpb.RecordCiphersuite_RECORD_CIPHERSUITE_AES_128_GCM_SHA256,
+									commonpb.RecordCiphersuite_RECORD_CIPHERSUITE_AES_256_GCM_SHA384,
+									commonpb.RecordCiphersuite_RECORD_CIPHERSUITE_CHACHA20_POLY1305_SHA256,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			description: "Get TLS config for server.",
+			request: &s2av2pb.SessionReq {
+				AuthenticationMechanisms: []*s2av2pb.AuthenticationMechanism {
+					{
+						MechanismOneof: &s2av2pb.AuthenticationMechanism_Token{"token"},
+					},
+				},
+				ReqOneof: &s2av2pb.SessionReq_GetTlsConfigurationReq {
+					&s2av2pb.GetTlsConfigurationReq {
+						ConnectionSide: commonpb.ConnectionSide_CONNECTION_SIDE_SERVER,
+					},
+				},
+			},
+			expectedResponse: &s2av2pb.SessionResp {
+				Status: &s2av2pb.Status {
+					Code: 0,
+				},
+				RespOneof: &s2av2pb.SessionResp_GetTlsConfigurationResp {
+					GetTlsConfigurationResp: &s2av2pb.GetTlsConfigurationResp {
+						TlsConfiguration: &s2av2pb.GetTlsConfigurationResp_ServerTlsConfiguration_{
+							&s2av2pb.GetTlsConfigurationResp_ServerTlsConfiguration{
+								CertificateChain: []string{
+									string(serverCert),
+								},
+								MinTlsVersion: commonpb.TLSVersion_TLS_VERSION_1_3,
+								MaxTlsVersion: commonpb.TLSVersion_TLS_VERSION_1_3,
+								HandshakeCiphersuites: []commonpb.HandshakeCiphersuite{},
+								RecordCiphersuites: []commonpb.RecordCiphersuite {
+									commonpb.RecordCiphersuite_RECORD_CIPHERSUITE_AES_128_GCM_SHA256,
+									commonpb.RecordCiphersuite_RECORD_CIPHERSUITE_AES_256_GCM_SHA384,
+									commonpb.RecordCiphersuite_RECORD_CIPHERSUITE_CHACHA20_POLY1305_SHA256,
+								},
+								TlsResumptionEnabled: false,
+								RequestClientCertificate: s2av2pb.GetTlsConfigurationResp_ServerTlsConfiguration_REQUEST_AND_VERIFY,
+								MaxOverheadOfTicketAead: 0,
+							},
+						},
+					},
+				},
+			},
+		},
+	}{
+		t.Run(tc.description, func(t *testing.T) {
+			// Start up Fake S2Av2 Server.
+			var wg sync.WaitGroup
+			wg.Add(1)
+			address, stop, err := startFakeS2Av2Server(&wg)
+			wg.Wait()
+			// Create stream to server.
```

Comment:
nit: Please add a blank line between the `wg.Wait()` line and the comment line, just to break up the text a little bit. Similarly below.